var/spack/repos/builtin/packages/haproxy/package.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Haproxy(MakefilePackage):
+    """
+    HAProxy is a single-threaded, event-driven, non-blocking engine
+    combining a very fast I/O layer with a priority-based scheduler.
+    """
+
+    homepage = "https://www.haproxy.org"
+    url      = "https://www.haproxy.org/download/2.1/src/haproxy-2.1.0.tar.gz"
```

Comment:
You'll probably want to add a `url_for_version` so that versions other than 2.1.X can be downloaded in the future.

Reply:
fix it in https://github.com/spack/spack/pull/14160/commits/3a642884eadf6afb6de4148afdca87410fb459e5